route/v1/stable.route.testsuite.yaml

```
@@ -82,3 +82,538 @@ tests:
+          termination: passthrough
+        to:
+          kind: Service
+          name: hello-openshift
+        httpHeaders:
+          actions:
+            request:
+            - name: Content-Location
+              action:
+                type: Set
+                set:
+                  value: /my-first-blog-post
+            - name: X-SSL-Client-Cert 
+              action:
+                type: Set
+                set:
+                  value: "%{+Q}[ssl_c_der,base64]"
+            - name:   Content-Language
+              action:
+                type: Delete
+            - name: X-Target
+              action:
+                type: Set
+                set:
+                  value: "%[req.hdr(host),lower]"
+    expectedError: "header actions are not permitted when tls termination is passthrough."
+  - name: Should not allow to set/delete HSTS header.
+    initial: |
+      apiVersion: route.openshift.io/v1
+      kind: Route
+      metadata:
+        labels:
+          type: sharded
+        name: hello-openshift-edge-hsts
+        namespace: hello-openshift
+      spec:
+        subdomain: hello-openshift
+        tls:
+          termination: edge
+        to:
+          kind: Service
+          name: hello-openshift
+        httpHeaders:
+          actions:
+            response:
+            - name: X-Frame-Options
+              action:
+                type: Set
+                set:
+                  value: DENY
+            - name: Strict-Transport-Security
+              action:
+                type: Delete
+            request:
+            - name: Content-Location
+              action:
+                type: Set
+                set:
+                  value: /my-first-blog-post
+            - name:   Content-Language
+              action:
+                type: Delete
+    expectedError: "strict-transport-security header may not be set or deleted via header actions"
+  - name: Should not allow to set/delete HSTS header.
+    initial: |
+      apiVersion: route.openshift.io/v1
+      kind: Route
+      metadata:
+        labels:
+          type: sharded
+        name: hello-openshift-edge-hsts
+        namespace: hello-openshift
+      spec:
+        subdomain: hello-openshift
+        tls:
+          termination: edge
+        to:
+          kind: Service
+          name: hello-openshift
+        httpHeaders:
+          actions:
+            response:
+            - name: X-Frame-Options
+              action:
+                type: Set
+                set:
+                  value: DENY
+            - name: Proxy
+              action:
+                type: Delete
+            request:
+            - name: Content-Location
+              action:
+                type: Set
+                set:
+                  value: /my-first-blog-post
+            - name:   Content-Language
+              action:
+                type: Delete
+    expectedError: "proxy header may not be set or deleted via header actions"
```

The name doesn't match what the test actually does.  Also, as I said before, we don't care about proxy in the request headers, and we don't care if the route deletes the proxy header; we only need to block setting proxy in the request headers.
```suggestion
  - name: Should not allow to set proxy request header.
    initial: |
      apiVersion: route.openshift.io/v1
      kind: Route
      metadata:
        name: hello-openshift-edge-proxy
        namespace: hello-openshift
      spec:
        subdomain: hello-openshift
        tls:
          termination: edge
        to:
          kind: Service
          name: hello-openshift
        httpHeaders:
          actions:
            request:
            - name: Proxy
              action:
                type: Set
                value: example.xyz
    expectedError: "proxy header may not be set or deleted via header actions"
```

---

done